src/ai/ai_help_structs.cc

```diff
@@ -756,7 +773,7 @@ void ManagementData::mutate(const uint8_t pn) {
 		probability = kNoAiTrainingMutation;
 	}
 
-	set_military_number_at(kMutationRatePosition, probability - 101);
+	set_military_number_at(kMutationRatePosition, probability / 10 - 101);
```

Comment:
after looking into the code of set_military_number_at it seems that passing on the probability her might be wrong as this function expects the new value for the magic number at this position not a probability of mutation.

Reply:
I will remove all this stuff about saving muration rate into magic/military number. Normal mutation rate would be 200, training one 500. I do  not think it has any significant impact (saving the number)

Reply:
Ok, all this stuff removed, so you will need to look at the diff once more. Things are simplified and I think it is good.